macrostep.el

```diff
@@ -708,17 +711,20 @@ Also removes the overlay from `macrostep-overlays'."
     ;; Remove overlay from the list and delete it
     (setq macrostep-overlays
           (delq overlay macrostep-overlays))
+    (let ((back-overlay (overlay-get overlay 'back-overlay)))
+      (when back-overlay (delete-overlay back-overlay)))
     (delete-overlay overlay)))
 
 (defun macrostep-collapse-overlays-in (start end)
   "Collapse all macrostepper overlays that are strictly between START and END.
 
 Will not collapse overlays that begin at START and end at END."
   (dolist (ol (overlays-in start end))
-    (if (and (> (overlay-start ol) start)
-	     (< (overlay-end ol) end)
-	     (overlay-get ol 'macrostep-original-text))
-	(macrostep-collapse-overlay ol t))))
+    (when (and (overlay-buffer ol)        ; collapsing may delete other overlays
```

Comment:
Yeah, that's the bug. However, I only encountered the bug when using these extra highlight overlays. When I was testing I found that only these new overlays caused an error in that loop, though I can't be certain that it was never possible before.

Do you still want a separate commit for that part?